### ISSUE TITLE
ModuleCollection class not needed. 

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest#188]
I realize we can eliminate the ``ModuleCollection()`` class because all its doing is getting the command ``module restore`` and testing the collection works.

All we need to do is move the methods to ``Module`` class

```
class ModuleCollection:
    """Class declaration of ModuleCollection."""

    def __init__(self, collection, debug=True):
        """Initializer method of ModuleCollection class.
        :param collection: name of module collection
        :param debug: debug mode for troubleshooting
        :type collection: str
        :type debug: bool
        """
        self.debug = debug

        # raise TypeError exception if collection is not a string type since that is required
        # when working with module collection
        if not isinstance(collection, str):
            raise TypeError(f"Type Error: {collection} is not of type string")

        self.collection = collection
        self.module_cmd = f"module restore {self.collection}"

    def test_collection(self):
        """Test the module collection by running ``module restore <collection>``.
        :return: return code of ``module restore`` command
        :rtype: int
        """

        cmd = BuildTestCommand()
        cmd.execute(self.module_cmd)
        ret = cmd.returnCode()

        # print executed command for debugging
        if self.debug:
            print(f"[DEBUG] Executing module command: {self.module_cmd}")
            print(f"[DEBUG] Return Code: {ret}")

        return ret

    def get_command(self):
        """ Get the module command used to restore a collection
        :return: Return the actual command to restore a collection.
        :rtype: str
        """

        return self.module_cmd
```

We can rename ``get_command`` --> ``get_collection`` (open to better name)

So what it would look like from Module() class is the following

```
a = Module()
a.test_collection("gcc") # this will test module restore gcc
a.get_collection("gcc")  # this will return module restore gcc
```

The Module() class implements most of the collection operation it really does make sense to move this over.

``module save`` --> ``Module().save()``
``module describe`` --> ``Module().describe()``
``module restore <collection>`` --> ``Module().get_collection()``

I think this would make sense and avoid having a class ``ModuleCollection`` which serves no purpose once the methods are moved.

@vsoch if you agree, it should be fairly simple fix.

**Comment by vsoch** [comment]
I don't think the name of the class, "Module" is intuitive if it can describe one or more modules (or a collection), but I like the idea of a simpler implementation with a class. If this is in fact a collection of modules, we should call it `ModuleCollection` ? 

I think I know what is bothering me about the name - what is potentially confusing is that when we think of "module" we think of one of the module recipes (e.g., gcc) but here the context is better mirroring the module command. Can we make this more explicit, perhaps call it `ModuleCommand`?

**Comment by shahzebsiddiqui** [comment]
I am +1 for changing class **Module** --> **ModuleCommand**. This avoids the confusion between the actual command ``module`` and module collection and even the import name ``from buildtest.module import Module`` and the python ``module`` concept Yikes! Last but not least, there is buildtest own module collection system that can be invoked during build using ``buildtest build --module-collection``. This is an alternative to Lmod User Collection which is more flexible in handling many collections. See https://buildtest.readthedocs.io/en/latest/module_operation.html#module-collection-operation-buildtest-module-collection


According to Lmod the correct terminology is *user collections* that is to refer to a list of module by a collection name see https://lmod.readthedocs.io/en/latest/010_user.html#user-collections

I hope these definitions can clarify your confusion
- **module** - A wrapper to bash and csh/tcsh used for dynamically changing user environment (PATH, LD_LIBRARY_PATH, MANPATH, etc...) by loading one or more module files
- **Modulefile** - A LUA or tcl file containing instructions on how to load an application environment
- **User Collection** - a list of one or more modules that can be referenced by a collection name. The collection name is not an actual module. Collections are found in ``$HOME/.lmod.d/<collection>``.  A collection name **default** will be found in ``$HOME/.lmod.d/default``

**Comment by vsoch** [comment]
Yes! ModuleCommand is good! It makes it so much more clear. I’d even say we should have a documentation page that briefly overviews some high-level concepts and has links for more detail.

**Comment by shahzebsiddiqui** [comment]
@vsoch I will work on this fix and it should be fairly simple. Once it's done I will fix the regression test and eventually write documentation on the Module API

**Comment by shahzebsiddiqui** [comment]
Closing this issue now this was fixed in #192
